src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -44,31 +44,57 @@ const getInitialReportScreenParams = (reports, ignoreDefaultRooms) => {
     return {reportID: String(reportID)};
 };
 
-const MainDrawerNavigator = (props) => {
-    const initialParams = getInitialReportScreenParams(props.reports, !Permissions.canUseDefaultRooms(props.betas));
+class MainDrawerNavigator extends Component {
+    constructor(props) {
+        super(props);
 
-    // Wait until reports are fetched and there is a reportID in initialParams
-    if (!initialParams.reportID) {
-        return <FullScreenLoadingIndicator />;
+        this.state = {
+            currentlyViewedReportID: '',
+        };
+
+        this.setCurrentlyViewedReportID = this.setCurrentlyViewedReportID.bind(this);
     }
 
-    // After the app initializes and reports are available the home navigation is mounted
-    // This way routing information is updated (if needed) based on the initial report ID resolved.
-    // This is usually needed after login/create account and re-launches
-    return (
-        <BaseDrawerNavigator
-            drawerContent={() => <SidebarScreen />}
-            screens={[
-                {
-                    name: SCREENS.REPORT,
-                    component: ReportScreen,
-                    initialParams,
-                },
-            ]}
-            isMainScreen
-        />
-    );
-};
+    setCurrentlyViewedReportID(reportID) {
+        this.setState({currentlyViewedReportID: reportID});
+    }
+
+    render() {
+        const initialParams = getInitialReportScreenParams(this.props.reports, !Permissions.canUseDefaultRooms(this.props.betas));
+
+        // Wait until reports are fetched and there is a reportID in initialParams
+        if (!initialParams.reportID) {
+            return <FullScreenLoadingIndicator />;
+        }
+
+        // After the app initializes and reports are available the home navigation is mounted
+        // This way routing information is updated (if needed) based on the initial report ID resolved.
+        // This is usually needed after login/create account and re-launches
+        return (
+            <BaseDrawerNavigator
+                drawerContent={() => (
```

Comment:
> Avoid using arrow function callbacks in components that are expensive to re-render. React will re-render this component since each time the parent renders it creates a new instance of that function. Alternative: Bind the method in the constructor instead - https://github.com/Expensify/App/blob/main/PERFORMANCE.md#react-performance-tips

@mdneyazahmad Let's remove the arrow function

Reply:
We can not remove it because we have to pass `currentlyViewedReportID`. The other option is to use `Context`. May be, we can make `SidebarScreen` `PureComponent`.

Reply:
I need your help here. What do you think @rushatgabhane ?

Reply:
If I'm not wrong, making it a pure component won't help because we have an arrow function.

Anyway, regarding removing the arrow function. It might not make a difference. (I'll verify this once again) 

So let's do nothing here